index.js

```diff
@@ -24,74 +24,86 @@ function computeTextColor(uicolors) {
    *  if c <= 0.03928 then c = c/12.92 else c = ((c+0.055)/1.055) ^ 2.4
    *  L = 0.2126 * r + 0.7152 * g + 0.0722 * b
    */
-  var c = uicolors.map((c)=> {
-    c = Number(c);
-    if (c <= 0.03928 ) {
-      return c/12.92;
-    } else {
-      return Math.pow((c+0.055)/1.055, 2.4);
+  const c = uicolors.map((color) => {
+    const colorNum = Number(color);
+    if (colorNum <= 0.03928) {
+      return colorNum / 12.92;
     }
+    return ((colorNum + 0.055) / 1.055) ** 2.4;
   });
 
-  var L = 0.2126 * c[0] + 0.7152 * c[1] + 0.0722 * c[2];
+  const L = 0.2126 * c[0] + 0.7152 * c[1] + 0.0722 * c[2];
 
   return (L > Threshold) ? '#000000ff' : '#ffffffff';
   // well if that doesnt' work, simplify!
-  //colors.textColor = ((r*255*0.299 + g*255*0.587 + b*255*0.114) > 186) ? '#000000' : '#ffffff';
+  // colors.textColor = ((r*255*0.299 + g*255*0.587 + b*255*0.114) > 186) ? '#000000' : '#ffffff';
 }
+// eslint-disable-next-line consistent-return
 export const getNamedSwatches = (image, callback) => {
   if (Platform.OS === 'android') {
     return RNPalette.getNamedSwatches(image, callback);
-  } else {
-    callback("Not supported");
   }
-}
-
+  callback('Not supported');
+};
 
+// eslint-disable-next-line consistent-return
 export const getAllSwatches = (options, image, callback) => {
   if (Platform.OS === 'android') {
-    return RNPalette.getAllSwatches(image, callback);
+    return RNPalette.getAllSwatches(image, (err, res) => {
+      if (err) {
+        return callback(err);
+      }
+      res.sort((a, b) => b.population - a.population);
+      const splitColor = res[0].color.split(',');
+      const r = Number.parseFloat(splitColor[0].split('rgba(')[1]);
+      const g = Number.parseFloat(splitColor[1]);
+      const b = Number.parseFloat(splitColor[2]);
+      res[0].hex = `#${toHex(r)}${toHex(g)}${toHex(b)}${toHex(0.6)}`;
```

Comment:
@chetstone 
side note here: in my case I only need dominant hex, that's why in this Android case, only res[0] has hex property
Should we include hex property for all swatches?

Reply:
Yes, if we return a hex property it should be done everywhere.